pkg/config/expand.go

```diff
@@ -405,95 +375,26 @@ func (dc *DeploymentConfig) applyGlobalVariables() error {
 	})
 }
 
-/*
-A module reference is made by the use keyword and is subject to IGC constraints
-of references (ordering, explicitness). It has the following fields:
-  - toModuleID: the target module ID
-  - fromModuleID: the source module ID
-  - toGroupID: the deployment group in which the module is *expected* to be found
-  - fromGroupID: the deployment group from which the reference is made
-  - explicit: a boolean value indicating whether the user made a reference that
-    explicitly identified toGroupID rather than inferring it using fromGroupID
-*/
-type modReference struct {
-	toModuleID   string
-	fromModuleID string
-	toGroupID    string
-	fromGroupID  string
-}
-
-/*
-This function performs only the most rudimentary conversion of an input
-string into a modReference struct as defined above. This function does not
-ensure the existence of the module!
-*/
-func identifyModuleByReference(yamlReference string, bp Blueprint, fromMod string) (modReference, error) {
-	// struct defaults: empty strings and false booleans
-	var ref modReference
-	ref.fromModuleID = fromMod
-	ref.toModuleID = yamlReference
-
-	fromG, err := bp.ModuleGroup(fromMod)
-	if err != nil {
-		return modReference{}, err
-	}
-	ref.fromGroupID = fromG.Name
-
-	toG, err := bp.ModuleGroup(ref.toModuleID)
-	if err != nil {
-		return modReference{}, err
-	}
-	ref.toGroupID = toG.Name
-
-	// should consider more sophisticated definition of valid values here.
-	// for now check that no fields are the empty string; due to the default
-	// zero values for strings in the "ref" struct, this will also cover the
-	// case that modComponents has wrong # of fields
-	if ref.fromModuleID == "" || ref.toModuleID == "" || ref.fromGroupID == "" || ref.toGroupID == "" {
-		return ref, fmt.Errorf("%s: %s, expected %s",
-			errorMessages["invalidMod"], yamlReference, expectedModFormat)
-	}
-
-	return ref, nil
-}
-
 // AutomaticOutputName generates unique deployment-group-level output names
 func AutomaticOutputName(outputName string, moduleID string) string {
 	return outputName + "_" + moduleID
 }
 
-func (ref modReference) validate(bp Blueprint) error {
-	callingModuleGroupIndex := slices.IndexFunc(bp.DeploymentGroups, func(d DeploymentGroup) bool { return d.Name == ref.fromGroupID })
-	if callingModuleGroupIndex == -1 {
-		return fmt.Errorf("%s: %s", errorMessages["groupNotFound"], ref.fromGroupID)
-	}
-
-	targetModuleGroupIndex, err := modToGrp(bp.DeploymentGroups, ref.toModuleID)
+func validateModuleReference(bp Blueprint, from Module, toID string) error {
+	to, err := bp.Module(toID)
 	if err != nil {
 		return err
```

Comment:
I believe this block will become self-explanatory (no comment needed), if you change the func name to `GetModule` as advised elsewhere. Otherwise, this block needs a comment like:

```go
// error if reference target module not found
```